include/painless/parameter.h

```diff
@@ -19,7 +20,13 @@
 
 namespace painless {
 
-static constexpr const char* BASE_PATH = "/tmp/painless/";
+inline std::string get_base_path() {
+  const char *tmpdir = nullptr;
+  if ((tmpdir = getenv("TMPDIR")) == nullptr) {
+      tmpdir = "/tmp";
+  }
+  return tmpdir + std::string{"/painless/"};
+}
```

Comment:
In principle this could be made allocation-free by using a static buffer of size `PATH_MAX`.

Reply:
We could also compute the filename just once during `Parameter<T>` construction. This way, we would not repeatedly call `get_base_path` later on. In any case, the path is only needed during parameter creation and in the watcher-thread, when a file is actually changed. Both parts are not really performance-critical (In my opinion, only `Parameter::value()` is performance critical).